src/test/java/tennisclub/entity/UserTest.java

```diff
@@ -0,0 +1,61 @@
+package tennisclub.entity;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+public class UserTest {
```

Comment:
I don't think this was even needed as it's not in the requirements _(testing DAO was enough)_ but good job for testing `equals` and `hashCode`👍 although I'm rather confused by the behaviour of the `equals` method - so if the two instances are "initialized" _(first setting of the username)_ with a same username, they are equal, otherwise they are not? That's some really dark magic going on here 😁

Reply:
@Fuyunori 

The correct implementation of `equals()` and `hashCode()` seems to be a recurring confusion for all of us. :laughing: My reasoning is as follows: `username` is a unique (both logically and by a DB constraint) attribute, i.e. 2 users with the same username must, in fact, be the same user. Since no other attributes are marked as unique (although email probably could be), `username` is also **the only** criterion usable for equality checking.

Do you think the reasoning is incorrect?

Reply:
I get the logic, what was confusing me was the fact that once you change the username of both users _(who shared the same username at the start)_ to something different, they are still equal, which is rather unexpected😅but thanks for the explanation though, I really appreciate it 👍

Reply:
I'm sorry, I was blind xD the username is set only once, I misread that as consecutive setting of the username 😅😅😅